api/envoy/admin/v2alpha/tap.proto

```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+
+import "envoy/service/tap/v2alpha/common.proto";
+import "validate/validate.proto";
+
+package envoy.admin.v2alpha;
+option java_package = "io.envoyproxy.envoy.admin.v2alpha";
+option java_multiple_files = true;
+
+// The /tap admin request body that is used to configure an active tap session.
+message TapRequest {
+  // The opaque configuration ID used to match the configuration to a loaded extension.
```

Comment:
The `TapRequest` can itself contain the full tap specification as long as there is  a single tap filter?

Reply:
The request does take the full tap specification. Are you saying to not make the ID required and fail if there is ambiguity on which filter to target?

Reply:
Think many listeners in the server, each of which has a tap filter installed. This is likely to be very common, for example, ingress and egress, etc.

Reply:
True, OK, this makes sense. The other alternative is to make this less opaque and allow specification of the particular listener, e.g. by listener index.

Reply:
Yeah I suppose it could be by listener name (I prefer that over index since it's easier to understand), though this ID allows a name to span multiple listeners which IMO is kind of nice. I think I will keep this unless you think otherwise.

Reply:
Fair enough, just wanted to make sure all design points were covered.